### PR TITLE
bugfix: ngx.re: fixed a split() edge-case when using control characters in the regex.

### PR DESCRIPTION
**Pull request by thibaultcha**
> I hereby granted the copyright of the changes in this pull request
to the authors of this lua-resty-core project.

Proposed fix to openresty/lua-nginx-module#1217.

### CODE REVIEW

File: lib/ngx/re.lua [hunk 1]
```diff
@@ -153,6 +160,9 @@ function _M.split(subj, regex, opts, ctx, max, res)
     local res_idx = 0
     local last_empty_match
 
+    -- update to split_helper indexes
+    ctx.pos = ctx.pos - 1
```
Comment (agentzh) [inline lib/ngx/re.lua]: Should we use `sub_idx` here on the RHS to avoid a duplicate `ctx.pos` table lookup?

File: lib/ngx/re.lua [hunk 2]
```diff
@@ -87,11 +87,18 @@ local function re_split_helper(subj, compiled, compile_once, flags, ctx)
         return nil, nil, nil
     end
 
+    if from == to then
+      -- empty match, skip to next char
+      ctx.pos = to + 1
+
+    else
+      ctx.pos = to
```
Comment (agentzh) [inline lib/ngx/re.lua]: Please use 4-space indentations consistently.

Comment (thibaultcha) [reply lib/ngx/re.lua]: Sorry, pushed this fix from an usual setup while on holidays, will update.

File: lib/ngx/re.lua (hunk 1 above)
Comment (thibaultcha) [reply lib/ngx/re.lua]: Fine with me, updated.

**Comment by thibaultcha** [comment]
@agentzh Ping for 1.13.8.1, I don't think we should leave bugfixes behind :/ I'm hoping this API will finally stabilize now...

**Comment by agentzh** [comment]
@thibaultcha Merged. Thanks!

**Pull request closed without merging**